TeamCode/src/main/java/com/acmerobotics/relicrecovery/vision/JewelTracker.java

```diff
@@ -0,0 +1,119 @@
+package com.acmerobotics.relicrecovery.vision;
+
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
+
+import org.opencv.core.Core;
+import org.opencv.core.Mat;
+import org.opencv.core.Rect;
+import org.opencv.core.Scalar;
+
+/**
+ * Created by ryanbrott on 9/23/17.
+ */
+
+public class JewelTracker implements Tracker {
+    public static final String TAG = "JewelTracker";
+
+    public static final float JEWEL_PLATFORM_ASPECT_RATIO = 2.6f; // width/height
+    public static final double COLOR_THRESHOLD = 0.7;
+
+    private Paint paint;
+    private Rect jewelPlatformRect, leftJewelRect, rightJewelRect;
+    private double leftRed, leftBlue, rightRed, rightBlue;
+
+    public JewelTracker() {
+        paint = new Paint();
+        paint.setStyle(Paint.Style.STROKE);
+        paint.setStrokeWidth(5.0f);
+    }
+
+    @Override
+    public synchronized void processFrame(Mat frame, double timestamp) {
+        if (jewelPlatformRect == null) {
+            int imageWidth = frame.width(), imageHeight = frame.height();
+            int platformWidth = imageWidth / 2;
```

Comment:
Is this measured from wherer we know the camera is going to be, or will it need to be updated once we get the robot?

Reply:
This represents the location of the jewel platform in the starting image. Yes, it will have to be updated once the robot is finished (or potentially determined by fancy transformations).